src/frontend/src/handler/create_table.rs

```diff
@@ -1261,9 +1287,261 @@ pub async fn handle_create_table(
     );
 
     let catalog_writer = session.catalog_writer()?;
-    catalog_writer
-        .create_table(source, table, graph, job_type)
-        .await?;
+
+    // Handle engine
+    match engine {
```

Comment:
The create table statement is not atomic. It creates hummock table, iceberg sink and iceberg source one by one. It needs to be improved in the future.